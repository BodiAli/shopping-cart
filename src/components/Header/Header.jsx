import { NavLink } from "react-router-dom";
import styles from "./Header.module.css";

function Header() {
  return (
    <>
      <header className={styles.header}>
        <h1>GameVault</h1>
        <div className={styles.navContainer}>
          <NavLink to="/">Home</NavLink>
          <NavLink to="shop">Shop</NavLink>
          <NavLink to="about">About</NavLink>
        </div>
      </header>
    </>
  );
}

export default Header;
