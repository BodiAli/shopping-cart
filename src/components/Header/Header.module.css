.header {
  background-color: var(--primary);
  padding: 1.5rem 1rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.header > h1 {
  font-family: "Merriweather", Times, serif;
  font-size: 2.2rem;
  font-weight: bold;
  background-image: url(/green-bg-text.jpg);
  background-clip: text;
  background-position: center;
  filter: brightness(1.8);
  color: transparent;
}

.navContainer {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 100px;
}

.navContainer > a {
  text-decoration: none;
  color: var(--darkerGreen);
  padding: 0.6rem 1.2rem;
  border-radius: 5px;
  background-color: var(--darkerGray);
}
