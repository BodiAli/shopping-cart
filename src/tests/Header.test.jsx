import { render, screen } from "@testing-library/react";
import Header from "../components/Header/Header";
import { expect } from "vitest";

describe("Header component", () => {
  test("should render correct elements", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  test("should render header element", () => {
    render(<Header />);
    expect(screen.getByRole("heading")).toBeInTheDocument();
  });

  test("should render home button", () => {
    render(<Header />);
    expect(screen.getByRole("button", { name: "Home" })).toBeInTheDocument();
  });
});
