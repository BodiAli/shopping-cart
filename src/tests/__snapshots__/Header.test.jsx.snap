// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > should render correct elements 1`] = `<div />`;
